src/main/webapp/resources/js/pages/analysis/components/tree/Tree.jsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * This file renders the tree component
+ */
+
+import React, { useContext, useEffect, useState } from "react";
+import { getI18N } from "../../../../utilities/i18n-utilities";
+import { TabPaneContent } from "../../../../components/tabs/TabPaneContent";
+import Phylocanvas from "react-phylocanvas";
+import { Button, Radio } from "antd";
+import { AnalysisContext } from "../../../../contexts/AnalysisContext";
+import { getNewickTree } from "../../../../apis/analysis/analysis";
+import { ContentLoading } from "../../../../components/loader/ContentLoading";
+import { WarningAlert } from "../../../../components/alerts/WarningAlert";
+import styled from "styled-components";
+import { SPACE_MD, SPACE_XS } from "../../../../styles/spacing";
+import { BORDER_LIGHT } from "../../../../styles/borders";
+
+const CANVAS_HEIGHT = 600;
+
+const VisualizationWrapper = styled.div`
+  height: ${CANVAS_HEIGHT}px;
+  border: ${BORDER_LIGHT};
+`;
+
+const ButtonGroupWrapper = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin-bottom: ${SPACE_MD};
+`;
+
+export default function Tree() {
+  const [newickString, setNewickString] = useState(null);
+  const [serverMsg, setserverMsg] = useState(null);
+  const [currTreeShape, setCurrTreeShape] = useState("circular");
+  const { analysisContext } = useContext(AnalysisContext);
+
+  // On load gets the newick string for the analysis
+  useEffect(() => {
+    getNewickTree(analysisContext.analysis.identifier).then(data => {
+      if (data.newick === null) {
+        //Empty tree
+        setNewickString("");
+      } else {
+        setNewickString(data.newick);
+      }
+
+      if (data.message !== null) {
+        setserverMsg(data.message);
+      }
+    });
+  }, []);
+
+  function getTree() {
+    return (
+      <Phylocanvas
+        data={newickString}
+        treeType={currTreeShape}
+        style={{ height: "100%" }}
+      />
+    );
+  }
+
+  function handleClick(e) {
+    setCurrTreeShape(e.target.value);
+    getTree();
```

Comment:
Instead of calling `getTree()` here you should use an effect that watches for the current tree shape to change value

Reply:
Updated in fb39d9edc0b52faad1f3a62d2990c41b8b6ca3ed